### ISSUE TITLE
about number of features in prediction 

### ISSUE THREAD

**Issue by Madr75** [QData/spacetimeformer#10]
Thank you for great well organized project!

Looking to the data processing functions you set features (target_cols variable) the same for X and for y. Can you please recommend "how" if I want to predict one feature from many in the dataset how to pass it to models?

**Comment by jakegrigsby** [comment]
Hmm this is interesting. I can't think of a reason why this wouldn't work conceptually, but none of our experiments had a Multivariate --> Univariate format so the loss functions are not built for it. I am adding this to the list of new experiments for version 2 of the paper. In the meantime (if you're fine wasting lot of compute) you could probably hack this together by padding the target values with a magic number and then using the missing value masking that's already present. The one value you're interested in predicting is there and the rest of the dimensions have a number that will be masked by calling forecaster.set_null_value. https://github.com/QData/spacetimeformer/blob/efff49aea19eb9d6f6643aafa4916aa387893d84/spacetimeformer/forecaster.py#L66-L81 So your targets and inputs would both be shape (x, num_target_cols) but the targets would have num_target_cols-1 columns of say 0.0. The forecaster then has its null value set to 0.0 and keep predictions on those values from impacting the loss function, meaning only the predictions on the real target variable are trained.

Otherwise, I'm planning on continuing work on a version 2 of this paper and codebase in a few weeks. Hopefully this will be released late winter (Feb or March)
